ansible/django-ansible-base
# Encrypt `extra_data` to not expose things that might be used by authenticators

When a user logs in through an authenticator we capture information about the login into a field called extra_data. The authenticator (ldap/keycloak/github/any others) could add sensitive information which is then plain text in the database. 

This PR encrypts that data before it goes in to the database and includes test coverage for those functions.

I also included an update to the requirements file for the dev environment as they are needed for testing/migration making for test app. I can split those out into a separate PR if needed. Django is pinned to the same versions that are required for regular install. 

# Testing Steps
1. Add any authenticator you wish to the API. Easiest is just enable ldap via the dev stack in the container start up file and un comment the `ldap_enabled: True` line. 
2. login in with a user via the authenticator. 
3. open a dbshell `docker exec -it aap_gw_1 aap-gateway-manage $* dbshell`
4. select the extra_data column from authenticatoruser (this will return any authenticatorUser in the db, change the query if you wish to select a specific one but _everything_ should be encrypted for `extra_data`) `select extra_data from dab_authentication_authenticatoruser;`
5. confirm the field starts with `$(encrypted)`
